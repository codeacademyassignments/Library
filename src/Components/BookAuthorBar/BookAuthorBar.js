import React, { Component } from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import style from './BookAuthorBar.style';

export default class BookAuthorBar extends Component {
  getAuthorComponents = (authorArray) => {
    const authorTextArray = authorArray.slice(0, 2).reduce((accumulator, current) => {
      accumulator.push(<Text key={current} style={style.author}>{current}</Text>);
      return accumulator;
    }, []);
    return authorTextArray;// limiting to two
  }

  render() {
    const { authorArray } = this.props;
    return (
      <View style={style.authorNames}>
        {this.getAuthorComponents(authorArray)}
      </View>
    );
  }
}

BookAuthorBar.propTypes = {
  authorArray: PropTypes.array.isRequired,
};
