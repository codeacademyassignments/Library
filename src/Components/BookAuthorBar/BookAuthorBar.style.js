import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  authorNames: {
    fontWeight: '100',
    flex: 1,
    flexDirection: 'row',
  },
  author: {
    fontSize: 10,
    marginRight: 10,
  },
});
