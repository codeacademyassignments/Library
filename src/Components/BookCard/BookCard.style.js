import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  image: {
    flex: 1,
    paddingTop: 0,
    marginTop: 0,
    maxHeight: 200,
  },
  title: {
    fontSize: 20,
    fontWeight: '900',
  },
  authorNames: {
    height: 15,
  },
  subtitle: {
    marginTop: 10,
    fontSize: 15,
    fontWeight: '700',
  },
  description: {
    fontSize: 12,
    marginTop: 10,
  },
  bookCard: {
    flex: 1,
    flexDirection: 'row',
    marginBottom: 10,
    height: 200,
  },
  bookDetails: {
    flex: 1,
    height: 200,
    justifyContent: 'space-between',
    paddingLeft: 10,
    paddingRight: 20,
    overflow: 'hidden',
  },
});
