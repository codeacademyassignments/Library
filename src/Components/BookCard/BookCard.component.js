import React, { Component } from 'react';
import {
  View, Text, Image, ScrollView, Alert,
} from 'react-native';
import PropTypes from 'prop-types';
import style from './BookCard.style';
import BookAuthorBar from '../BookAuthorBar/BookAuthorBar';
import defaultBookImage from '../../Images/defaultBook.jpg';
// import defaultBook from '../../Images/defaultBook.jpg';

export default class BookCard extends Component {
  // createBookAuthorBar = authors => (
  //   (authors !== undefined && authors !== null)
  //     ? (<BookAuthorBar authorArray={authors} />)
  //     : (<></>)
  // );

  // addSubtitle = subtitle => (
  //   (subtitle !== undefined && subtitle !== null)
  //     ? (<Text adjustsFontSizeToFit numberOfLines={2} style={style.subtitle}>{subtitle}</Text>)
  //     : <></>
  // )


  // addImage = imageLinks => (
  //   (imageLinks === undefined || imageLinks === null || imageLinks.thumbnail === undefined)
  //     ? (
  //       <Image style={style.image} resizeMode="contain" source={deafultBookImage} />)
  //     : (
  //       <Image style={style.image} resizeMode="contain" source={{ uri: imageLinks.thumbnail }} />
  //     ))


  render() {
    const { book } = this.props;
    const { title, imageLinks } = book;
    let {
      subtitle, authors, description,
    } = book;
    subtitle = subtitle || '';
    authors = authors || [];
    description = description || '';
    const imageSource = imageLinks ? { uri: imageLinks.thumbnail } : defaultBookImage;
    return (
      <View style={style.bookCard}>
        <Image style={style.image} resizeMode="contain" source={imageSource} />
        <View style={style.bookDetails}>
          <View>
            <Text adjustsFontSizeToFit numberOfLines={2} style={style.title}>{title}</Text>
            <View style={style.authorNames}>{<BookAuthorBar authorArray={authors} />}</View>
          </View>
          <Text adjustsFontSizeToFit numberOfLines={2} style={style.subtitle}>{subtitle}</Text>
          <View style={style.description}><ScrollView><Text adjustsFontSizeToFit>{description}</Text></ScrollView></View>
        </View>
      </View>
    );
  }
}

BookCard.propTypes = {
  book: PropTypes.shape({
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    authors: PropTypes.array,
    description: PropTypes.string,
    imageLinks: PropTypes.shape({
      thumbnail: PropTypes.string,
    }),
  }).isRequired,
};

// BookCard.defaultProps = {
//   book: {
//     subtitle: 'anything',
//     authors: [],
//     description: 'something more here',
//   },
// };
