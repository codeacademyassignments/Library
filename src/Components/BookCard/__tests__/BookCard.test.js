import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Image, Text } from 'react-native';
import BookAuthorBar from '../../BookAuthorBar/BookAuthorBar';
import style from '../BookCard.style';
import BookCard from '../BookCard.component';
import deafultBookImage from '../../../Images/defaultBook.jpg';

const testBookData = {
  title: 'APJ',
  authors: ['J K Rowling'],
  description: 'was great person',
  imageLinks: { thumbnail: 'path/to/file' },
};

describe('BookCard', () => {
  it('should match with old snapshot', () => {
    const tree = renderer.create(<BookCard book={{
      title: testBookData.title, authors: testBookData.authors, description: testBookData.description,
    }}
    />);
    expect(tree).toMatchSnapshot();
  });
});

// describe('addImage', () => {
//   it('should return Image component with default image source', () => {
//     const wrapper = shallow(<BookCard book={{ title: testBookData.title, authors: testBookData.authors, description: testBookData.description }} />);
//     expect(wrapper.instance().addImage(undefined)).toEqual(<Image style={style.image} resizeMode="contain" source={deafultBookImage} />);
//   });
//   it('should return Image component with passed image source', () => {
//     const wrapper = shallow(<BookCard book={{
//       ...testBookData,
//     }}
//     />);
//     expect(wrapper.instance().addImage({ thumbnail: 'path/to/file' })).toEqual(<Image style={style.image} resizeMode="contain" source={{ uri: 'path/to/file' }} />);
//   });
// });


// describe('addSubtitle', () => {
//   it('should not add subtitle when subtitle is undefined or null', () => {
//     const wrapper = shallow(<BookCard book={{
//       ...testBookData,
//     }}
//     />);
//     expect(wrapper.instance().addSubtitle()).toEqual(<></>);
//   });
//   it('should add subtitle when subtitle is some string', () => {
//     const wrapper = shallow(<BookCard book={{
//       ...testBookData,
//     }}
//     />);
//     const subtitle = 'About Great Persons';
//     expect(wrapper.instance().addSubtitle(subtitle)).toEqual(<Text adjustsFontSizeToFit numberOfLines={2} style={style.subtitle}>{subtitle}</Text>);
//   });
// });

// describe('createBookAuthorBar', () => {
//   it('should not create book author bar when undefined or null passed', () => {
//     const wrapper = shallow(<BookCard book={{
//       ...testBookData,
//     }}
//     />);
//     expect(wrapper.instance().createBookAuthorBar()).toEqual(<></>);
//   });
//   it('should add subtitle when subtitle is not undefined or null', () => {
//     const wrapper = shallow(<BookCard book={{
//       ...testBookData,
//     }}
//     />);
//     const authors = ['J K rowling', 'APJ', 'Paulo Coelo'];
//     expect(wrapper.instance().createBookAuthorBar(authors)).toEqual(<BookAuthorBar authorArray={authors} />);
//   });
// });
