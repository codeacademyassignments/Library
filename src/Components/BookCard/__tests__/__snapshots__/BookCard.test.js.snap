// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard should match with old snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "height": 200,
      "marginBottom": 10,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/Images/defaultBook.jpg",
      }
    }
    style={
      Object {
        "flex": 1,
        "marginTop": 0,
        "maxHeight": 200,
        "paddingTop": 0,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "height": 200,
        "justifyContent": "space-between",
        "overflow": "hidden",
        "paddingLeft": 10,
        "paddingRight": 20,
      }
    }
  >
    <View>
      <Text
        adjustsFontSizeToFit={true}
        numberOfLines={2}
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "900",
          }
        }
      >
        APJ
      </Text>
      <View
        style={
          Object {
            "height": 15,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "fontWeight": "100",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 10,
                "marginRight": 10,
              }
            }
          >
            J K Rowling
          </Text>
        </View>
      </View>
    </View>
    <Text
      adjustsFontSizeToFit={true}
      numberOfLines={2}
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "700",
          "marginTop": 10,
        }
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "fontSize": 12,
          "marginTop": 10,
        }
      }
    >
      <RCTScrollView>
        <View>
          <Text
            adjustsFontSizeToFit={true}
          >
            was great person
          </Text>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
